compiler/bindings/c/iree/compiler/dialects/iree_gpu.h

```diff
@@ -150,12 +150,19 @@ struct ireeGPUTargetInfo {
   MlirAttribute maxWorkgroupSizes;    // Max threads per X/Y/Z dimension.
   int64_t maxThreadCountPerWorkgroup; // Max threads per workgroup.
   int64_t maxWorkgroupMemoryBytes;    // Max workgroup memory.
+  MlirAttribute mmaIntrinsics;        // MMA Intrinsics.
 };
 
 // Queries GPU target info from the given `ExecutableTargetAttr` attribute.
 MLIR_CAPI_EXPORTED ireeGPUTargetInfo
 ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr);
 
+MLIR_CAPI_EXPORTED ireeGPUTargetInfo ireeGPUTargetInfoGet(
+    MlirContext mlirCtx, const char *arch, const int32_t *subgroupChoices,
+    size_t numSubgroupChoices, const int32_t *workgroupSizes,
+    size_t numWorkgroupSizes, int64_t threadCount, int64_t memoryBytes,
+    const int32_t *mmaIntrinsics, size_t numMmaIntrinsics);
```

Comment:
How do you know `mmaIntrinsics` map to `int32_t`?

Reply:
How can I tell it's i32 instead of u32? Also, what happens if this changes in the future? I think we need to have a typedef on the C api side, static_assert in the C implementation, and then use this typedef on the python side.